src/router/middlewarePipeline.js

```diff
@@ -0,0 +1,18 @@
+function middlewarePipeline(context, middleware, index) {
```

Comment:
A cleaner version would be to have all the logic in one place, let's keep the index.js just for the routes
```
let loginRoute = {
  path: '/login'
};

function redirectToLogin(next) {
  return next(loginRoute);
}

/**
 * Middleware to check if user has the correct right to access a certain page.
 * Checks are performed based on route meta fields `meta: { requiresAuth: true }`.
 * @param {object} router Vue router instance
 */
export default function authMiddleware(router) {
  router.beforeEach(async (to, from, next) => {

    const requiresAuth = to.matched.some(record => record.meta.requiresAuth);

    if (!requiresAuth) {
      return next();
    }

    const token = localStorage.getItem('token');

    if (!token) {
      return redirectToLogin(next);
    }
    return next();
  });
}
```

Reply:
Then in main.js you import this function
`import authMiddleware from "@/router/authMiddleware";`
Then you pass it the router
`authMiddleware(router)`

Reply:
The `record.meta.requiresAuth` is set in the apiConfig file
```
function isUnauthorized(status) {
  const isOnSecuredRoute = router.app.$route.matched.some(record => record.meta.requiresAuth)

  return [401, 403].includes(status) && isOnSecuredRoute
}
```

Reply:
```
export async function errorInterceptor(error) {
    const { status } = error.response;
    let errors = '';

    if (statusCodesToHandle.includes(status)) {
        errors = mapErrors(error.response.data);
    }

    error.errors = errors;

    if (isUnauthorized(status)) {
    
        await AuthService.logout();
        await app.$router.push('/login');
    
        return Promise.reject(error);
      }

    return Promise.reject(error);
}
```

Reply:
Let me know if it works 👍

Reply:
it works :)